Source/Core/Core/AchievementManager.cpp

```diff
@@ -987,6 +994,38 @@ void AchievementManager::FetchBadge(AchievementManager::Badge* badge, u32 badge_
       if (name_to_fetch.empty())
         return;
     }
+
+    std::string cache_base = File::GetUserPath(D_RETROACHIEVEMENTSCACHE_IDX);
+    auto context = Common::SHA1::CreateContext();
+    UpdateFromArray(*context, Common::BitCastToArray<u8>(badge_type));
+    context->Update(std::vector<u8>(name_to_fetch.begin(), name_to_fetch.end()));
+    auto name_hash = context->Finish();
```

Comment:
`badge_type` is an integer so you don't need to hash it, all possible values are easy to prove that they can be converted into valid and unique strings. I'd just use `Common::SHA1::CalculateDigest(name_to_fetch.data(), name_to_fetch.size())` here.

Reply:
It at least needs to be appended to the name because, for example, the locked and unlocked version of the same achievement will likely have the same base name and are only differentiated by the badge type. Without the badge type the names are no longer unique.

Reply:
Yeah I get that part, I was thinking of something along the lines of:

```
    const std::string cache_path =
        fmt::format("{}/{}-{}.png", cache_base, badge_type,
                    Common::SHA1::ToString(Common::SHA1::CalculateDigest(name_to_fetch)));
```

Reply:
Ohh, got it.